ops/testing.py

```diff
@@ -1141,6 +1141,67 @@ def reset_planned_units(self):
         """
         self._backend._planned_units = None
 
+    def add_network(self, binding_name: str, address: str, *,
+                    relation_id: Optional[int] = None,
+                    cidr: Optional[str] = None,
+                    interface: str = 'eth0',
+                    ingress_addresses: Optional[List[str]] = None,
```

Comment:
Could this accept a broader type of `Optional[typing.Iterable[str]]`? This would mean that the interface could accept more types giving the caller more flexibility. E.g., if the ingress_addresses don't change, the user may want to use a tuple or use a generator to conserve memory. Similar comment for `egress_subnets`

Reply:
Fair point -- updated.